__tests__/melt-tokens.test.js

```diff
@@ -0,0 +1,67 @@
+import TestUtils from "./test-utils";
+
+// TODO: Add a fixture to melt token 01
+describe("melt-tokens api", () => {
+  it("should return 200 with a valid body", async () => {
+    const response = await TestUtils.request
+      .post("/wallet/melt-tokens")
+      .send({
+        token: "00da712d64e04866c8c9aa8fceca70e80d1693864176b6b443220cf29adab5ed",
+        amount: 1,
+      })
+      .set({ "x-wallet-id": TestUtils.walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.hash).toBeDefined();
+  });
+
+  it("should melt a token with amount as string", async () => {
+    const response = await TestUtils.request
+      .post("/wallet/melt-tokens")
+      .send({
+        token: "00da712d64e04866c8c9aa8fceca70e80d1693864176b6b443220cf29adab5ed",
+        amount: "1",
+      })
+      .set({ "x-wallet-id": TestUtils.walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.hash).toBeDefined();
+  });
+
+  it("should not melt a token without the required parameters", async () => {
+    ["token", "amount"].forEach(async (field) => {
+      const token = {
+        token: "00da712d64e04866c8c9aa8fceca70e80d1693864176b6b443220cf29adab5ed",
+        amount: 1,
+      };
+      delete token[field];
+      const response = await TestUtils.request
+        .post("/wallet/melt-tokens")
+        .send(token)
+        .set({ "x-wallet-id": TestUtils.walletId });
+      expect(response.status).toBe(400);
+      expect(response.body.success).toBeFalsy();
+    });
+  });
+
+  it("should receive an error when trying to do concurrent melt-tokens (lock/unlock behavior)", async () => {
```

Comment:
I feel we should test melting less than 1 token, so there won't be any HTR in return of the melt operation. It was a bug in previous versions of the wallet.

Reply:
Amount 1 is actually 0.01, so I guess it's already tested.

I think we should add a test with 0 amount, to validate it won't execute.

And add test with 100, to see the return deposit.

Reply:
You're right. Thanks!

Reply:
1. `I think we should add a test with 0 amount, to validate it won't execute.`
Done
2. `And add a test with 100, to see the return deposit.`
I`m unable to test it without mocking the `fullnode` response, but doing by it I will be only testing a mocked response.